File: Studio.API/Studio.API/Studio.API/Business/Domain/Results/Locations/CityResult.cs
using Studio.API.Business.Domain.Results.Bases;
using System.ComponentModel.DataAnnotations.Schema;
namespace Studio.API.Business.Domain.Results.Locations
{
    public class CityResult : BaseResult
    {
        public string CityName { get; set; }
        
        public Guid CountryId { get; set; }

        public CountryResult Country { get; set; }
        
        public IList<LocationResult> Locations { get; set; }
    }
}

File: Studio.UI/src/app/data/results/locations/CityResult.ts
import { Guid } from "guid-typescript";
import { BaseResult } from "../base/BaseResult";
import { LocationResult } from "./LocationResult";
import { CountryResult } from "./CountryResult";

export interface CityResult extends BaseResult{
    cityName: string;
    countryId: Guid;
    country: CountryResult;
    locations: LocationResult[];
}